docs/rules/sort-comp.md

```diff
@@ -171,6 +172,48 @@ var Hello = React.createClass({
 });
 ```
 
+If you want to flow annotations to be at the top:
+
+```js
+"react/sort-comp": [1, {
+  order: [
+    'type-annotations',
+    'static-methods',
+    'lifecycle',
+    'everything-else',
+    'render',
+  ],
+}]
+```
+
+With the above configuration, the following patterns are considered warnings:
+
+```js
+class Hello extends React.Component<any, Props, void> {
+  onClick() { this._someElem = true; }
+  props: Props;
```

Comment:
It gets completely removed altogether!

http://babeljs.io/repl/#?evaluate=true&lineWrap=false&presets=es2015%2Ces2015-loose%2Creact%2Cstage-0&experimental=true&loose=true&spec=false&code=type%20Props%20%3D%20%7B%20a%3A%20number%20%7D%3B%0Aclass%20B%20%7B%0A%20%20props%3A%20Props%3B%0A%20%20%0A%7D